### PR TITLE
[Merged by Bors] - feat: explicit limits in Profinite

### PR DESCRIPTION
**Pull request by dagurtomas**
Co-authored by:

- Riccardo Brasca [riccardo.brasca@gmail.com](mailto:riccardo.brasca@gmail.com) @riccardobrasca
- Filippo A. E. Nuccio [filippo.nuccio@univ-st-etienne.fr](mailto:filippo.nuccio@univ-st-etienne.fr) @faenuccio

Analogue of CompHaus/ExplicitLimits for Profinite.
This work was done during the 2023 Copenhagen masterclass on formalisation of condensed mathematics. Numerous participants contributed.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
Please remember to add the label `awaiting-review` when this is ready for review. (Alternatively PRs that refactor files that were ported from mathlib3 should be labelled `after-port`.)

File: Mathlib/Topology/Category/Profinite/ExplicitLimits.lean
Comment (riccardobrasca) [inline Mathlib/Topology/Category/Profinite/ExplicitLimits.lean:21]:
```suggestion
This is essentially just copied from the `CompHaus` folder
```

File: Mathlib/Topology/Category/Profinite/ExplicitLimits.lean
Comment (riccardobrasca) [inline Mathlib/Topology/Category/Profinite/ExplicitLimits.lean:149]: I feel it's more natural to state the reversed equality, as it is done below for the coproduct. I don't have a real argument, so if you prefer this order I'm fine with it, but it should be the same for all explicit constructions.

**Review by riccardobrasca — APPROVED** [review]

File: Mathlib/Topology/Category/Profinite/ExplicitLimits.lean
Comment (dagurtomas) [reply Mathlib/Topology/Category/Profinite/ExplicitLimits.lean:117]: I want to have the explicit stuff on the LHS, and the abstract limit stuff on the RHS. This approach is now unified in this PR and the corresponding one for CompHaus

File: Mathlib/Topology/Category/Profinite/ExplicitLimits.lean
Comment (semorrison) [inline Mathlib/Topology/Category/Profinite/ExplicitLimits.lean:1]: Could this file just be called `Limits.lean`? I think that would be consistent with naming elsewhere.

File: Mathlib/Topology/Category/Profinite/ExplicitLimits.lean
Comment (semorrison) [inline Mathlib/Topology/Category/Profinite/ExplicitLimits.lean:106]: Should not start with an uppercase.

Perhaps rename to just `isoPullback`? Or `pullbackIsoPullback`?

File: Mathlib/Topology/Category/Profinite/ExplicitLimits.lean
Comment (semorrison) [inline Mathlib/Topology/Category/Profinite/ExplicitLimits.lean:182]: Similarly here.

File: Mathlib/Topology/Category/Profinite/ExplicitLimits.lean
Comment (semorrison) [inline Mathlib/Topology/Category/Profinite/ExplicitLimits.lean:191]: And here.

File: Mathlib/Topology/Category/Profinite/Limits.lean
Comment (TwoFX) [inline Mathlib/Topology/Category/Profinite/Limits.lean:191]: It feels inconsistent to have `coproductHomeoCoproduct` but not `pullbackHomeoPullback`. Unless you have some immediate use for it, I would just remove `coproductHomeoCoproduct` for now, considering that you don't develop any API for it. If it turns out to be useful later, it can still be added again together with the necessary API.

File: Mathlib/Topology/Category/Profinite/Limits.lean
Comment (TwoFX) [inline Mathlib/Topology/Category/Profinite/Limits.lean:106]: There is a missing docstring here (which CI does not catch since the corresponding linter is still disabled in CI).

File: Mathlib/Topology/Category/Profinite/Limits.lean
Comment (TwoFX) [inline Mathlib/Topology/Category/Profinite/Limits.lean:182]: Another missing docstring.

File: Mathlib/Topology/Category/Profinite/Limits.lean
Comment (dagurtomas) [reply Mathlib/Topology/Category/Profinite/Limits.lean:191]: It will be used in upcoming PR's developing condensed mathematics. But I can add `pullbackHomeoPullback` for consistency.

**Comment by TwoFX** [comment]
Thanks! Please wait until CI has passed before merging.
bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**
